### PR TITLE
[VEN-2557] Update XVSVault Implementation

### PR DESCRIPTION
**Pull request by 0xlucian**
## Description

This PR introduces VIPs and Simulations for updating the `XVSVault` Implementations on the following chains:

1. `Opbnbtestnet` & `Opbnbmainnet`
2. `Bsctestnet` & `Bscmainnet`
3. `Sepolia` & `Ethereum`
4. `Arbitrumsepolia`

The simulations check for storage layout collision by reading directly from storage the pending withdrawals before upgrade and after upgrade calling the public function for same rewards and comparing the results.

**NOTE:** For `OpBNBTestnet` and `OpBNBMainnet` no pending rewards were found so the value read there is 0.
**NOTE:** Need to update address of new implementation for Ethereum.

Resolves VEN-2557

### CODE REVIEW

File: multisig/simulations/opbnbtestnet/vip-018/simulations.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { expect } from "chai";
+import { BigNumber, Contract, utils } from "ethers";
+import { ethers } from "hardhat";
+
+import { calculateMappingStorageSlot } from "../../../../src/utils";
+import { forking, pretendExecutingVip } from "../../../../src/vip-framework";
+import vip018 from "../../../proposals/opbnbtestnet/vip-018";
+import XVSVault_ABI from "./abi/XVSVault_ABI.json";
+
+const XVS_VAULT_PROXY = "0xe4DE4460c93Af011DC0624bc7Ce4a2Bf40feE4ea";
```
Comment (Debugger022) [inline multisig/simulations/opbnbtestnet/vip-018/simulations.ts:10]: This address belongs to XVS_VAULT_IMPLEMENTATION. XVS_VAULT_PROXY is at 
`0xB14A0e72C5C202139F78963C9e89252c1ad16f01`

File: hardhat.config.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ const BLOCK_GAS_LIMIT_PER_NETWORK = {
   bsctestnet: 140000000,
   bscmainnet: 140000000,
   sepolia: 30000000,
+  arbitrumsepolia: 30000000,
```
Comment (Debugger022) [inline hardhat.config.ts:24]: It is already added.
```suggestion
```

**Comment by Debugger022** [comment]
We should once check that the permission of `setRewardAmountPerBlockOrSecond` is given.

File: multisig/simulations/ethereum/vip-021/simulations.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  let pendingWithdrawalsBefore: BigNumber;
+
+  before(async () => {
+    xvsVaultProxy = new ethers.Contract(XVS_VAULT_PROXY, XVSVault_ABI, provider);
+    const storageSlot = calculateMappingStorageSlot(XVS_ADDRESS, POOL_ID, MAPPING_STORAGE_SLOT);
+    const value = await provider.getStorageAt(XVS_VAULT_PROXY, storageSlot);
+    pendingWithdrawalsBefore = BigNumber.from(utils.stripZeros(value));
+  });
+
+  describe("Post-VIP behavior", async () => {
```
Comment (GitGuru7) [inline multisig/simulations/ethereum/vip-021/simulations.ts:27]: I would add a basic sanity check for the new implementation address

Comment (GitGuru7) [inline multisig/simulations/ethereum/vip-021/simulations.ts:27]: Can we add a generic check for XVSVault from `src/vip-framework/checks/checkXVSVault.ts`

File: vips/vip-292/bscmainnet.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      {
+        target: XVS_VAULT_PROXY,
+        signature: "initializeTimeManager(bool,uint256)",
+        params: [false, BNB_BLOCKS_PER_YEAR],
+      },
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlockOrSecond(address,uint256)", bscmainnet.NORMAL_TIMELOCK],
+      },
```
Comment (chechu) [inline vips/vip-292/bscmainnet.ts:43]: We should allow to execute `setRewardAmountPerBlockOrSecond` also to the Fast-track and Critical timelocks

Comment (chechu) [reply vips/vip-292/bscmainnet.ts:43]: Also on bsctestnet

File: multisig/proposals/opbnbmainnet/vip-020/index.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    {
+      target: NEW_XVS_IMPLEMENTATION,
+      signature: "_become(address)",
+      params: [XVS_VAULT_PROXY],
+    },
+    {
+      target: XVS_VAULT_PROXY,
+      signature: "initializeTimeManager(bool,uint256)",
+      params: [true, 0],
```
Comment (chechu) [inline multisig/proposals/opbnbmainnet/vip-020/index.ts:24]: `opBNB` is block based, with 1 block per second

File: multisig/proposals/opbnbtestnet/vip-018/index.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    {
+      target: NEW_XVS_IMPLEMENTATION,
+      signature: "_become(address)",
+      params: [XVS_VAULT_PROXY],
+    },
+    {
+      target: XVS_VAULT_PROXY,
+      signature: "initializeTimeManager(bool,uint256)",
+      params: [true, 0],
```
Comment (chechu) [inline multisig/proposals/opbnbtestnet/vip-018/index.ts:25]: It's block based, with 1 block per second

File: multisig/proposals/opbnbmainnet/vip-020/index.ts (hunk 5 above)
Comment (Debugger022) [reply multisig/proposals/opbnbmainnet/vip-020/index.ts:24]: the first parameter will be false too as it is block-based on opbnb network not time-based.

File: multisig/proposals/opbnbtestnet/vip-018/index.ts (hunk 6 above)
Comment (Debugger022) [reply multisig/proposals/opbnbtestnet/vip-018/index.ts:25]: the first parameter will be false too as it is block-based on opbnb network not time-based.

File: vips/vip-292/bscmainnet.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlockOrSecond(address,uint256)", FAST_TRACK_TIMELOCK],
+      },
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlockOrSecond(address,uint256)", CRITICAL_TIMELOCK],
+      },
+    ],
```
Comment (chechu) [inline vips/vip-292/bscmainnet.ts:57]: We should revoke the permission to invoke `setRewardAmountPerBlock`:

```
      {
        target: ACM,
        signature: "revokeCallPermission(address,string,address)",
        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlock(address,uint256)", FAST_TRACK_TIMELOCK],
      },
      {
        target: ACM,
        signature: "revokeCallPermission(address,string,address)",
        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlock(address,uint256)", CRITICAL_TIMELOCK],
      },
      {
        target: ACM,
        signature: "revokeCallPermission(address,string,address)",
        params: [XVS_VAULT_PROXY, "setRewardAmountPerBlock(address,uint256)", NORMAL_TIMELOCK],
      },
```

**Review by GitGuru7 — APPROVED** [review]

**Comment by Debugger022** [comment]
Opbnbtestnet tx: https://testnet.opbnbscan.com/tx/0x8bec3dc30e1452df56bc9b5a3b99464749478880f404c74f021b3f2280304dde
Sepolia tx: https://sepolia.etherscan.io/tx/0x54944627ff5a79a02d303775620795df49a9ad4784c96f979b9e66d338ef9db6
Bsctestnet vip: https://testnet.venus.io/#/governance/proposal/409?chainId=97
ArbitrumSepolia tx: https://sepolia.arbiscan.io/tx/0xdae099b6db87a7958046ab12bad45f5cd77923f2c4f7641d1936aacf2535b74e

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/314?chainId=56

**Pull request merged**